### PR TITLE
[JENKINS-50726] Form validations

### PR DESCRIPTION
**Pull request by kuisathaverat**
[JENKINS-50726](https://issues.jenkins-ci.org/browse/JENKINS-50726)

* Form validations

follow up of https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/35 the only change is https://github.com/jenkinsci/artifact-manager-s3-plugin/commit/c1373684c3c9d4c1835265a24b9fad204281aa31

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStore.java [hunk 1]
```diff
@@ -187,12 +211,80 @@ public URL toExternalURL(@NonNull Blob blob, @NonNull HttpMethod httpMethod) thr
...
+            }
+            return ret;
+        }
+
+        public FormValidation doCheckPrefix(@QueryParameter String prefix){
+            FormValidation ret = FormValidation.ok();
+            if(StringUtils.isBlank(prefix)){
+                ret = FormValidation.error("Prefix should have a value");
+            } else if(!prefix.endsWith("/")){
+                ret = FormValidation.warning("if Prefix point to a folder, it should end with '/' character");
```
Comment (carlossg) [inline src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStore.java]: I think it's better to always add it if not present, better UX

Comment (kuisathaverat) [reply src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStore.java]: but you could use it as refix, I mean, if you set "folder/prefix-", all the stuff will be created as "folder/prefix-WHATEVER" that could be the expected behavior.
If this is not the expected behavior the warning will change to "a '/' will be added to the end"

Comment (carlossg) [reply src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStore.java]: ok, so you are saying that `folder/prefix-` is a valid value and just want to print a warning about it. Fare enough

File: src/main/resources/io/jenkins/plugins/artifact_manager_s3/S3BlobStore/config.jelly [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
+    <f:section title="${%Artifact Manager S3}">
+        <j:if test="${!descriptor.isPropertyConfigured()}">
+            <f:entry title="${%S3 Bucket Name}" field="container">
+                <f:textbox/>
+            </f:entry>
+            <f:entry title="${%S3 Bucket Region}" field="region">
+                <f:select/>
+            </f:entry>
+            <f:entry title="${%Base folder}" field="prefix">
```
Comment (carlossg) [inline src/main/resources/io/jenkins/plugins/artifact_manager_s3/S3BlobStore/config.jelly]: should not use "folder", prefix then

File: src/main/resources/io/jenkins/plugins/artifact_manager_s3/S3BlobStore/config.jelly [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        </j:if>
+        <j:if test="${descriptor.isPropertyConfigured()}">
+            <strong>Configured by properties - Read Only</strong>
+            <f:entry title="${%S3 Bucket Name}">
+                <f:readOnlyTextbox value="${descriptor.container}"/>
+            </f:entry>
+            <f:entry title="${%S3 Bucket Region}">
+                <f:readOnlyTextbox value="${descriptor.region}"/>
+            </f:entry>
+            <f:entry title="${%Base folder}">
```
Comment (carlossg) [inline src/main/resources/io/jenkins/plugins/artifact_manager_s3/S3BlobStore/config.jelly]: same as above

**Pull request closed without merging**